incubator/hnc/internal/validators/object.go

```diff
@@ -142,6 +148,24 @@ func validateSelectorChange(inst, oldInst *unstructured.Unstructured) error {
 	return err
 }
 
+func validateTreeSelectorChange(inst, oldInst *unstructured.Unstructured) error {
+	oldSelectorStr := selectors.GetTreeSelectorAnnotation(oldInst)
+	newSelectorStr := selectors.GetTreeSelectorAnnotation(inst)
+	if newSelectorStr == "" || oldSelectorStr == newSelectorStr {
+		return nil
+	}
+	errStrs := validation.IsDNS1123Label(newSelectorStr)
+	if len(errStrs) != 0 {
+		return errors.New(newSelectorStr + ", " + strings.Join(errStrs, ". "))
+	}
+	_, err := selectors.GetTreeSelector(inst)
```

Comment:
The error message should be generated in the `selectors` package, not here, so that we can display the _same_ error from the reconciler if needed. `GetTreeSelector` should be the one that returns the error.

Reply:
done